### ISSUE TITLE
[ban-types] Does not ban {}

### ISSUE THREAD

**Issue by mulholio** [typescript-eslint/typescript-eslint#1185]
I want to ban the `{}` type. Unfortunately, this is not working.

**Repro**

```javascript
// .eslintrc.js
module.exports = {
  "rules": {
    '@typescript-eslint/ban-types': [
      'error',
      {
        types: {
          '{}': {
            message:
              'Every type except null and undefined is assignable to {}. Use object instead',
            fixWith: 'object',
          },
          Object: {
            message:
              'Every type except null and undefined is assignable to Object. Use object instead',
            fixWith: 'object',
          },
        },
      },
    ],
};
```

```TS
const bannedObject: Object = 2; // Error, as expected
const goodObject: object = {}; // No error, as expected

const missedObject: {} = 2; // No error, unexpected!
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^1.13.0` |
| `@typescript-eslint/parser`        | `^1.13.0` |
| `TypeScript`                       | `^3.5.3` |
| `ESLint`                           | `^5.16.0` |
| `node`                             | `X.Y.Z` |
| `npm`                              | `X.Y.Z` |

**Comment by bradzacher** [comment]
That's because `{}` is not a named type, which is what the rule operates on.
Happy to accept a PR to add a special case for this particular annotation.
